Added BZs and known issues to rel notes

bfallonf opened the pull request:
@CowboysFan If you want to take a look at the Rel Notes and make any suggestions.

CowboysFan commented:
@bfallonf Overall it looks good, but here are my initial thoughts on formatting and style:

1. Let's make "For Admins" and "For Developers" section headings that don't appear in the TOC; I'm pretty sure you can do that.

2. I don't like the way the text rolls under the BZ links in the Bugs section. I'm also not sure if they need to be bullet points if we can find some other way to do it. For example, a 2-column table with no borders at all, which would keep the two columns aligned vertically. There may also be another way, but I'd look into that because I think it's a bit messy the way it presents now.

Let me know if you have any questions.

adellape commented:
@CowboysFan @bfallonf 

> We need short headings that then describe those features/fixes in more detail and context.

For 3.0 GA's "New Features", we did this using a labeled list (http://asciidoctor.org/docs/user-manual/#labeled-list), which achieves the same effect:

- Source: https://github.com/openshift/openshift-docs/blame/master/whats_new/ose_3_0_release_notes.adoc#L28-L51

- docs.openshift: https://docs.openshift.com/enterprise/3.0/whats_new/ose_3_0_release_notes.html#new-features

- Portal: https://access.redhat.com/documentation/en/openshift-enterprise/version-3.0/openshift-enterprise-30-whats-new/#new-features

IMO, doing it this way (just speaking to this Rel Note usage, really) is cleaner than a bunch of manually-crafted bold "headings".

> Let's make "For Admins" and "For Developers" section headings that don't appear in the TOC; I'm pretty sure you can do that. Then, each section under there should also be a heading or bold text. This applies to other sections as well.

Not sure if you're looking at only the GH preview or a real build, but the way Brice has them now (which we used for errata notes for 3.0.1 and 3.0.2), "For Administrators" and "For Developers" are actually titles of the two lists themselves. Pretty build (internal) of Brice's PR:

- http://file.rdu.redhat.com/~adellape/111315/bfallon-relnotes2/release_notes/ose_3_1_release_notes.html#ose-31-new-features-and-enhancements

Here's the same in use in, for example, the 3.0.1 notes:

- https://docs.openshift.com/enterprise/3.0/whats_new/ose_3_0_release_notes.html#ose-3-0-1-0

And on the Portal, it comes out looking like what you're describing:

- https://access.redhat.com/documentation/en/openshift-enterprise/version-3.0/openshift-enterprise-30-whats-new/#ose-3-0-1-0

> The text rolling underneath the BZ links in the Bugs section doesn't look good.

What does this mean? Just that each BZ summary is in normal paragraph form?

> I'm also not sure if they need to be bullet points if we can find some other way to do it. For example, a 2-column table with no borders at all, which would keep the two columns aligned vertically. There may also be another way, but I'd look into that because I think it's a bit messy the way it presents now.

You can achieve this with a horizontal labeled list (see `[horizontal]` in http://asciidoctor.org/docs/user-manual/#labeled-list).

adellape commented on release_notes/ose_3_1_release_notes.adoc:
```diff
@@ -42,6 +42,13 @@ including any additional steps that may be required for a specific release.
 ====
 
 .For Administrators:
+* Several services and packages have been renamed:
```
This mentions only "services and packages", but the list also includes renamed directory and file names.

bfallonf replied:
Fixed

adellape commented on release_notes/ose_3_1_release_notes.adoc:
```diff
@@ -160,7 +164,22 @@ command to SSH into privileged containers.
 `iptables` command wait to acquire the *xtables* lock, causing some SDN
-initializations to fail. This issue has now been fixed.
+initializations to fail. This issue has been fixed.
+* https://bugzilla.redhat.com/show_bug.cgi?id=1272201[BZ 1272201]: When using two network interfaces, installing a clustered etcd and defining variables for IP and etcd interfaces, the certificate would be populated with only the first
```
I think there's a missing comma after "a clustered etcd". Hard to read without.

bfallonf replied:
Reworded

adellape commented on release_notes/ose_3_1_release_notes.adoc:
```diff
@@ -160,7 +164,22 @@ command to SSH into privileged containers.
+initializations to fail. This issue has been fixed.
+* https://bugzilla.redhat.com/show_bug.cgi?id=1272201[BZ 1272201]: When using two network interfaces, installing a clustered etcd and defining variables for IP and etcd interfaces, the certificate would be populated with only the first
+network, instead of whichever network was desired. The issue has now been fixed.
+* https://bugzilla.redhat.com/show_bug.cgi?id=1269256[BZ 1269256]: Using the `GET` fieldselector would return a 500 BadRequest error. This issue has been fixed.
```
Should be camelcase and marked up:

```
`*fieldSelector*`
```

adellape commented on release_notes/ose_3_1_release_notes.adoc:
```diff
@@ -160,7 +164,22 @@ command to SSH into privileged containers.
+* https://bugzilla.redhat.com/show_bug.cgi?id=1272201[BZ 1272201]: When using two network interfaces, installing a clustered etcd and defining variables for IP and etcd interfaces, the certificate would be populated with only the first
+network, instead of whichever network was desired. The issue has now been fixed.
+* https://bugzilla.redhat.com/show_bug.cgi?id=1269256[BZ 1269256]: Using the `GET` fieldselector would return a 500 BadRequest error. This issue has been fixed.
+* https://bugzilla.redhat.com/show_bug.cgi?id=1268000[BZ 1268000]: Previously, creating an application from a imagestream could result in two builds being initiated. This was caused by the wrong imagestream tag being used by the build process. The issue has been fixed.
```
s/imagestream/image stream/

Two of em.

adellape commented on release_notes/ose_3_1_release_notes.adoc:
```diff
@@ -160,7 +164,22 @@ command to SSH into privileged containers.
+network, instead of whichever network was desired. The issue has now been fixed.
+* https://bugzilla.redhat.com/show_bug.cgi?id=1269256[BZ 1269256]: Using the `GET` fieldselector would return a 500 BadRequest error. This issue has been fixed.
+* https://bugzilla.redhat.com/show_bug.cgi?id=1268000[BZ 1268000]: Previously, creating an application from a imagestream could result in two builds being initiated. This was caused by the wrong imagestream tag being used by the build process. The issue has been fixed.
+* https://bugzilla.redhat.com/show_bug.cgi?id=1267231[BZ 1267231]: The *ose-ha-proxy* router image was missing the x-forwarded header, causing the Jenkins application to redirect to HTTP instead of HTTPS. The issue has been fixed.
```
```
the `X-Forwarded` headers
```

Plural cuz there's multiple of them (`-Host`, `-Port`, `-Proto`).

adellape commented on release_notes/ose_3_1_release_notes.adoc:
```diff
@@ -160,7 +164,22 @@ command to SSH into privileged containers.
+* https://bugzilla.redhat.com/show_bug.cgi?id=1269256[BZ 1269256]: Using the `GET` fieldselector would return a 500 BadRequest error. This issue has been fixed.
+* https://bugzilla.redhat.com/show_bug.cgi?id=1268000[BZ 1268000]: Previously, creating an application from a imagestream could result in two builds being initiated. This was caused by the wrong imagestream tag being used by the build process. The issue has been fixed.
+* https://bugzilla.redhat.com/show_bug.cgi?id=1267231[BZ 1267231]: The *ose-ha-proxy* router image was missing the x-forwarded header, causing the Jenkins application to redirect to HTTP instead of HTTPS. The issue has been fixed.
+* https://bugzilla.redhat.com/show_bug.cgi?id=1276548[BZ 1276548]: Previously, an error was present where the HAProxy router did not expose stats, even if the port was specified. The issue has been fixed.
```
s/stats/statistics/

adellape commented on release_notes/ose_3_1_release_notes.adoc:
```diff
@@ -160,7 +164,22 @@ command to SSH into privileged containers.
+* https://bugzilla.redhat.com/show_bug.cgi?id=1267231[BZ 1267231]: The *ose-ha-proxy* router image was missing the x-forwarded header, causing the Jenkins application to redirect to HTTP instead of HTTPS. The issue has been fixed.
+* https://bugzilla.redhat.com/show_bug.cgi?id=1276548[BZ 1276548]: Previously, an error was present where the HAProxy router did not expose stats, even if the port was specified. The issue has been fixed.
+* https://bugzilla.redhat.com/show_bug.cgi?id=1275388[BZ 1275388]: Previously, some node hosts would not talk to the SDN due to routing table differences. A `*lbr0*` entry was causing traffic to be routed incorrectly. The issue has been fixed.
+* https://bugzilla.redhat.com/show_bug.cgi?id=1265187[BZ 1265187]: When persistent volume claims were created from a template, sometimes the same volume would be mounted to multiple persistent volume claims. At the same time, the volume would show that only one PVC was being used. The issue has been fixed.
```
"persistent volume claims (PVCs)" for the first one, then PVC for the two after?

adellape commented on release_notes/ose_3_1_release_notes.adoc:
```diff
@@ -160,7 +164,22 @@ command to SSH into privileged containers.
+* https://bugzilla.redhat.com/show_bug.cgi?id=1265187[BZ 1265187]: When persistent volume claims were created from a template, sometimes the same volume would be mounted to multiple persistent volume claims. At the same time, the volume would show that only one PVC was being used. The issue has been fixed.
+* https://bugzilla.redhat.com/show_bug.cgi?id=1279308[BZ 1279308]: Previously, using a etcd storage location other than the default, as defined in the master configuration file, would result in an upgrade fail at the "generate etcd backup" stage. This issue has now been fixed.
+* https://bugzilla.redhat.com/show_bug.cgi?id=1276599[BZ 1276599]: Basic authentication passwords can now contain colons.
+* https://bugzilla.redhat.com/show_bug.cgi?id=1279744[BZ 1279744]: Previously, giving EmptyDir volumes a different default permission setting and group ownership could affect deploying the *postgresql-92-rhel7* image. The issue has been fixed.
```
```
`*EmptyDir*`
```

adellape commented on release_notes/ose_3_1_release_notes.adoc:
```diff
@@ -160,7 +164,22 @@ command to SSH into privileged containers.
+* https://bugzilla.redhat.com/show_bug.cgi?id=1279744[BZ 1279744]: Previously, giving EmptyDir volumes a different default permission setting and group ownership could affect deploying the *postgresql-92-rhel7* image. The issue has been fixed.
+* https://bugzilla.redhat.com/show_bug.cgi?id=1276395[BZ 1276395]: Previously, an error could occur when trying to perform an HA install using Ansible, due to a problem with SRC files. The issue has been fixed.
+* https://bugzilla.redhat.com/show_bug.cgi?id=1267733[BZ 1267733]: When installing a etcd cluster with hosts with different network interfaces, the install would fail. The issue has been fixed.
+* https://bugzilla.redhat.com/show_bug.cgi?id=1274239[BZ 1274239]: Previously, when changing the default project region from infra to primary, old route and registry pods are stuck in the terminating stage and could not be deleted, meaning that new route and registry pods could not be deployed. The issue has been fixed.
```
`*infra* to *primary*`

adellape commented on release_notes/ose_3_1_release_notes.adoc:
```diff
@@ -160,7 +164,22 @@ command to SSH into privileged containers.
+* https://bugzilla.redhat.com/show_bug.cgi?id=1276395[BZ 1276395]: Previously, an error could occur when trying to perform an HA install using Ansible, due to a problem with SRC files. The issue has been fixed.
+* https://bugzilla.redhat.com/show_bug.cgi?id=1267733[BZ 1267733]: When installing a etcd cluster with hosts with different network interfaces, the install would fail. The issue has been fixed.
+* https://bugzilla.redhat.com/show_bug.cgi?id=1274239[BZ 1274239]: Previously, when changing the default project region from infra to primary, old route and registry pods are stuck in the terminating stage and could not be deleted, meaning that new route and registry pods could not be deployed. The issue has been fixed.
+* https://bugzilla.redhat.com/show_bug.cgi?id=1278648[BZ 1278648]: If, when upgrading to OpenShift Enterprise 3.1, the OpenShift repository was not set, a Python error would occur. This issue has been fixed.
```
s/OpenShift/OpenShift Enterprise/

adellape commented on release_notes/ose_3_1_release_notes.adoc:
```diff
@@ -175,4 +194,11 @@ Features Support Scope]
-- Binary builds, and the `*Dockerfile*` source type for builds.
+* Binary builds, and the Dockerfile source type for builds.
+* Support for OpenShift Enterprise running on Atomic Host.
+* Containerized installations, meaning all OpenShift Enterprise components running in a container.
```
More accurate to say "in containers".

adellape commented on release_notes/ose_3_1_release_notes.adoc:
```diff
@@ -175,4 +194,11 @@ Features Support Scope]
+
+== Known Issues
+
+* When pushing an internal registry when multiple registries share the same NFS volume, there is a chance the push will fail. A work-around has been suggested.
```
"When pushing _to_ an internal registry"? Maybe?

s/work-around/workaround/ (per IBM Style Guide)

Where is the suggested workaround?

bfallonf replied:
The workaround is #1174 

I was waiting for this to merge, but it's been a while. You and Paul Weil commented on (what looks like) your Friday, but I'll comment again. Either way, I was thinking this could link out to that, but as it's not merged yet, I got nothing to link to. I'll keep an eye on it.

bfallonf commented:
@CowboysFan @adellape OK. I made the lists into the labeled lists. I'll admit though, I did lik the way they looked on the portal...

Alex, addressed your comments. Pretty much just fixes, except for that last one which is a little more fiddley. I'll take another look tomorrow. Maybe #1174 will be merged by then...

If there's anything else, let me know. Thanks, guys!

miminar commented on release_notes/ose_3_1_release_notes.adoc:
```diff
@@ -42,125 +42,208 @@ including any additional steps that may be required for a specific release.
 environment.
-* Several persistent storage options are now available, such as Red Hat's
+
+More Persistant Storage Options::
```
`s/Persistant/Persistent/`

bfallonf replied:
Oops. Thanks @miminar

bfallonf commented on release_notes/ose_3_1_release_notes.adoc:
```diff
@@ -175,4 +258,12 @@ Features Support Scope]
 
-- Binary builds, and the `*Dockerfile*` source type for builds.
+* Binary builds, and the Dockerfile source type for builds.
+* Horizontally scaling application instances. OpenShift compares pod CPU usage and will scale appropriately depending on the `*TargetCPUConsumption*` parameter on the pod.
```
@DirectXMan12 Ah ok. I just had this info from a line in another document. I've fixed as suggested. Thanks!

bfallonf commented:
Ok. Gonna merge this. If there's anything else, we can make a follow-up PR.
